baremetal/metal3machine_manager_test.go

```diff
@@ -563,6 +594,96 @@ var _ = Describe("Metal3Machine manager", func() {
 		)
 	})
 
+	type testCaseSetPauseAnnotation struct {
+		BMMachine     *capm3.Metal3Machine
+		Host          *bmh.BareMetalHost
+		ExpectPresent bool
+		ExpectError   bool
+	}
+
+	DescribeTable("Test Set BMH Pause Annotation",
+		func(tc testCaseSetPauseAnnotation) {
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), tc.Host, tc.BMMachine)
+			machineMgr, err := NewMachineManager(c, nil, nil, nil, tc.BMMachine, klogr.New())
+			Expect(err).NotTo(HaveOccurred())
+
+			err = machineMgr.SetPauseAnnotation(context.TODO())
+			if tc.ExpectError {
+				Expect(err).To(HaveOccurred())
+			} else {
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			savedHost := bmh.BareMetalHost{}
+			err = c.Get(context.TODO(),
+				client.ObjectKey{
+					Name:      tc.Host.Name,
+					Namespace: tc.Host.Namespace,
+				},
+				&savedHost,
+			)
+			Expect(err).NotTo(HaveOccurred())
+			if tc.ExpectPresent {
+				Expect(savedHost.Annotations[bmh.PausedAnnotation]).NotTo(BeNil())
+			} else {
+				Expect(savedHost.Annotations[bmh.PausedAnnotation]).To(BeNil())
+				return
+			}
+
+		},
+		Entry("Set BMH Pause Annotation, with valid CAPM3 Paused annotations", testCaseSetPauseAnnotation{
+			Host: &bmh.BareMetalHost{
+				ObjectMeta: *bmhObjectMetaWithValidCAPM3PausedAnnotations(),
+				Spec: bmh.BareMetalHostSpec{
+					ConsumerRef: &corev1.ObjectReference{
+						Name:       "mybmmachine",
+						Namespace:  "myns",
+						Kind:       "BMMachine",
+						APIVersion: capm3.GroupVersion.String(),
+					},
+				},
+			},
+			BMMachine: newMetal3Machine("mybmmachine", nil, bmmSpec(), nil,
+				bmmObjectMetaWithValidAnnotations()),
+			ExpectPresent: true,
+			ExpectError:   false,
+		}),
+		Entry("Set BMH Pause Annotation, with valid Paused annotations, Empty Key", testCaseSetPauseAnnotation{
+			Host: &bmh.BareMetalHost{
+				ObjectMeta: *bmhObjectMetaWithValidEmptyPausedAnnotations(),
+				Spec: bmh.BareMetalHostSpec{
+					ConsumerRef: &corev1.ObjectReference{
+						Name:       "mybmmachine",
+						Namespace:  "myns",
+						Kind:       "BMMachine",
+						APIVersion: capm3.GroupVersion.String(),
+					},
+				},
+			},
+			BMMachine: newMetal3Machine("mybmmachine", nil, bmmSpec(), nil,
+				bmmObjectMetaWithValidAnnotations()),
+			ExpectPresent: true,
+			ExpectError:   false,
+		}),
+		Entry("Set BMH Pause Annotation, with no Paused annotations", testCaseSetPauseAnnotation{
+			Host: &bmh.BareMetalHost{
+				ObjectMeta: *bmhObjectMetaEmptyAnnotations(),
+				Spec: bmh.BareMetalHostSpec{
+					ConsumerRef: &corev1.ObjectReference{
+						Name:       "mybmmachine",
+						Namespace:  "myns",
+						Kind:       "BMMachine",
+						APIVersion: capm3.GroupVersion.String(),
+					},
+				},
+			},
+			BMMachine: newMetal3Machine("mybmmachine", nil, bmmSpec(), nil,
+				bmmObjectMetaWithValidAnnotations()),
+			ExpectPresent: true,
+			ExpectError:   false,
+		}),
+	)
+
```

Comment:
you're missing the test cases to remove the pause annotation

Reply:
Do you mean deletion? It is added later ...

Reply:
not deletion, no. the removal of the pause annotation when the cluster is unpaused

Reply:
Yeah, but I don't have the logic yet since I was unsure, how can I have tests :D

Reply:
Added